resources/config.ini

```diff
@@ -369,7 +369,14 @@ AcceptBidAskLastSizeDisplayUpdateNotification=
 
 SendMarketDataInLotsForUSstocks=
 
-
+# Sometimes the gateway runs into problems
+# and asks about diagnostics upload to IB.
+# 'yes' - allow to upload diagnostics
+# not set - ignore the dialog box.
+# The dialog is "harmless" and is not blocking any operation.
+# However setting "yes" would increase chances that IB fixes the problem.
+
+UploadIBDiagnosticsBundleIfAsked=

```

Comment:
This amendment is in entirely the wrong place. Please insert it at line 533 after the 'SaveTwsSettingsAt' setting.

Also the setting name is rather cumbersome.

I'd prefer to have the following wording (note that this uses some additional values for this setting, so further code updates will be needed):

```
# Upload Diagnostics on Request
# -----------------------------
#
# TWS may request diagnostics (which it generates) to be sent to IBKR
# developers in unexpected situations. The information may assist IBKR in
# diagnosing the issue. This setting tells IBC how to respond to the request.
#
# Permitted values are:
#
#    'send' - TWS sends the diagnostics to IBKR, and closes the request window
#
#    'sendwithscreenshot' - TWS sends the diagnostics, which include a full
#        screenshot of the desktop, to IBKR, and closes the request window. You
#        should consider carefully whether this screenshot might contain
#        personal information that you would not be comfortable passing to IBKR
#
#    'reject' - don't send diagnostics to IBKR, and close the request window
#
#    'ignore' (default value) - don't send diagnostics to IBKR, and leave the
#        request window on display for the user to handle. Note that this
#        window does not hinder normal TWS operation

DiagnosticsUploadAction=



```

Reply:
fixed

Reply:
sorry, I did one change. instead of DiagnosticsUploadAction I use TwsDiagnosticsUploadAction. Just to make sure it is clearly in the name that it is TWS not IBC asking for upload.

Reply:
Really I don't see any need for the `TWS` prefix. The comments make it quite clear what situation this setting relates to.

Reply:
renamed to DiagnosticsUploadAction